src/Tokenizer/Tokens.php

```diff
@@ -288,6 +288,36 @@ public function setSize($size)
         }
     }
 
+    /**
```

Comment:
The changes below are for SCA only right? I mean we would never actual get the value `null` from the parent? Maybe it would be better to throw a runtime exception like `Uncaught RuntimeException: Index invalid or out of range in /X/PHP-CS-Fixer/src/Tokenizer/Tokens.php:X` ? and not return an empty Token?

Reply:
[`SplFixedArray` returns `null` for unitinialized and unset indexes](https://3v4l.org/fRdPX). Throwing an exception would be a bigger BC break IMO, while returning an empty token matches the behavior that was documented and is consistent with what `clearAt()` method does.

Though I'm not against changing the behavior in 3.0.

Reply:
can we see the behavior, returning null, in the unit tests?

Reply:
I added tests.

Reply:
Thanks for the test :+1: 
Looking at it some more, I think updating the PHPDoc to `Token|null` is better than the empty token as returning the empty Token is a BC break IMHO. Also, on 3.0 it might be better to not allow calling the constructor of the Tokens class.

Reply:
[It's not *really* a BC break](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4970#issue-422465318). I agree the API of this class can be improved on 3.0 to prevent unexpected scenarios.